rplidar.go

```diff
@@ -40,19 +45,18 @@ var (
 type Rplidar struct {
 	resource.Named
 	resource.AlwaysRebuild
-	mu                      sync.Mutex
-	close                   bool
-	device                  rplidarDevice
-	nodes                   gen.Rplidar_response_measurement_node_hq_t
-	nodeSize                int
-	started                 bool
-	scannedOnce             bool
-	defaultNumScans         int
-	warmupNumDiscardedScans int
-
-	minRangeMM float64
-
-	logger logging.Logger
+
+	deviceMutex *sync.Mutex
```

Comment:
[note] two mutex now, one for device access and one for cache access. Both must be checked before Close can be done with cache needing to be checked first to prevent a deadlock.

Reply:
I believe this was an artifact from when I was experimenting with some testing and I was getting complaints about passing around a mutex by value.